storage/mock/mock.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2019 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// Package mock defines a fake storage implementation for use in testing.
```

Comment:
Can we move this under the internal package just for peace of mind? I don't see a reason to expose it outside of OPA proper for now.

Reply:
Also, once we update storage to error on read/writes on closed transactions some of this may change/go away.

Reply:
Moved in the latest update